text/0047-switch-to-advanced-docking-system.md

```diff
@@ -0,0 +1,302 @@
+
+Some hotkeys and pause functions rely on UI buttons, so OBS Basic will be refactored to remove this dependance.
+
+### Transistions widget
+*Future ADS Transitions dock widget*
+
+Transitions will be stored in OBSBasic and selected through the combo box of the widget.
+
+The transition duration will become an attribute of OBSBasic, and signal and slots will be setup to synchromise it with the dock spinbox.
+
+Buttons for adding, removing and open settings will be connected through signals and slots to OBSBasic.
+
+### Mixer widget
+*Future ADS Mixer dock widget*
+
+The base dock is easily separable, but the VolControl class can be refactored (toward MVC) later because this is not a requirement for this RFC.
+
+### Sources widget
+*Future ADS Sources dock widget*
+
+OBSBasic heavily rely on the SourceTree from the widget, but a refactoring will not be considered as a requirement for this RFC. So the class of the sources widget will have OBSBasic as a friend class.
+
+### Scenes widget
+*Future ADS Scenes dock widget*
+
+OBSBasic heavily rely on the SceneTree from the widget, but a refactoring will not be considered as a requirement for this RFC. So the class of the scenes widget will have OBSBasic as a friend class.
+
+### Central widget
+*Future ADS Central dock widget*
+
+Some context bar, preview/program and nudge related code is moved to the central widget class.
+
+This class will have OBSBasicPreview as a friend class.
+
+This class will be a friend class of OBSBasic also to have access to some gs_vertbuffer_t type attributes.
+
+## ADS
+The Advanced Docking System, require to add a Dock Manager in the main window and then add the central widget and then add other docks.
+
+Some work on themes CSS will be required.
+
+### About obs-deps
+
+obs-deps already build Qt for macOS and Windows, so adding a ADS script to the Qt tarballs is not a problem.
+
+### Floating docks titlebar on X11
+On X11, ADS provide two type of titlebar for floating docks:
+
+- Native
+
+[![Native titlebar under GNOME](./0047-switch-to-advanced-docking-system/linux_native_floating_dock_titlebar.png)]()
+
+- QWidget based (WIP CSS)
+
+[![QWidget titlebar](./0047-switch-to-advanced-docking-system/linux_qwidget_floating_dock_titlebar.png)]()
+
+The QWidget one is used by default when using KWin based Desktop Environement like Plasma. And this titlebar requires some CSS in the themes to match it.
+
+So the QWidget based titlebar usage will be enforced thanks to a flag for the Dock Manager to "unify" experience between desktop environement.
+Theme makers may ignore to theme this bar.
+
+### Dock state
+*The per profile dock state feature is not taken into account to describe those changes.*
+
+In the global config (`global.ini`).
+- `"dockState"` is kept for for backward compatiblity and no longer overwitten. It will be used if the following state is not present.
+- `"windowState"` is the state of the main window, since it does not store only the state of legacy docks.
```

Comment:
I assume this stores both legacy dock position, and main window position?

Reply:
Changed the text.